### PR TITLE
Adds set_pod_key action

### PR DESCRIPTION
**Pull request by marcelofabri**
Fixes https://github.com/fastlane/fastlane/issues/4408

### CODE REVIEW

File: fastlane/lib/fastlane/actions/set_pod_key.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          FastlaneCore::ConfigItem.new(key: :project,
+                                       env_name: "FL_SET_POD_KEY_PROJECT",
+                                       description: "The project name",
+                                       is_string: true,
+                                       optional: true)
+        ]
+      end
+
+      def self.is_supported?(platform)
+        return true
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: This should be `platform == :ios` 👍

Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: Or Mac too?

File: fastlane/lib/fastlane/actions/set_pod_key.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  module Actions
+    class SetPodKeyAction < Action
+      def self.run(params)
+        Actions.verify_gem!('cocoapods-keys')
+        cmd = []
+
+        cmd << ['bundle exec'] if File.exist?('Gemfile') && params[:use_bundle_exec]
+        cmd << ['pod keys set']
+
+        cmd << ["\"#{params[:key]}\""]
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: Please escape using `shellescape` here 👍

**Comment by KrauseFx** [comment]
Looks great, thanks for the PR 👍 I added one comment before it's ready to be merged

File: fastlane/lib/fastlane/actions/set_pod_key.rb (hunk 1 above)
Comment (marcelofabri) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: Mac too. Updated to make it explicit.

File: fastlane/lib/fastlane/actions/set_pod_key.rb (hunk 2 above)
Comment (marcelofabri) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: Nice catch!

**Comment by marcelofabri** [comment]
Done! ❤️

File: fastlane/lib/fastlane/actions/set_pod_key.rb [hunk 3]
```diff
@@ -8,9 +8,9 @@ def self.run(params)
         cmd << ['bundle exec'] if File.exist?('Gemfile') && params[:use_bundle_exec]
         cmd << ['pod keys set']
 
-        cmd << ["\"#{params[:key]}\""]
-        cmd << ["\"#{params[:value]}\""]
-        cmd << ["\"#{params[:project]}\""] if params[:project]
+        cmd << ["\"#{params[:key].shellescape}\""]
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: I'm not 100% certain, but I think when using `shellescape`, you shouldn't use the `"` around the string any more

**Comment by KrauseFx** [comment]
Looks great, thanks for updating. One more comment. Also, could you squash the commits to one? 👍

**Comment by KrauseFx** [comment]
Oh and I see we have conflicts 😢

File: fastlane/lib/fastlane/actions/set_pod_key.rb (hunk 3 above)
Comment (marcelofabri) [inline fastlane/lib/fastlane/actions/set_pod_key.rb]: But then wouldn't the tests break (each string would have two `"` around them)? I haven't touched then after changing this.

**Comment by marcelofabri** [comment]
Fixed the conflicts and rebased 💯

**Comment by KrauseFx** [comment]
This is great, thanks @marcelofabri 👍

**Comment by KrauseFx** [comment]
Awesome contribution, this will be in the next release 🚀

**Pull request merged**